blog/2024-02-09-private-api-gateway-aws.md

```diff
@@ -0,0 +1,113 @@
+
+You can check the CLI version like this:
+
+```bash
+wing --version
+0.57.16
+```
+
+> As you can see, Wing is still in pre-release, so expect some hiccups and don't be shy reporting issues or get help from the awesome people hanging out on the [Wing slack](https://t.winglang.io/slack).
+>
+
+Ok, now that we have the Wing CLI installed, let's create a new project using the `http-api` quickstart:
+
+```bash
+$ mkdir my-api
+$ cd my-api
+$ wing new http-api
+```
+
+---
+
+Let's check out what we now have in our project directory:
+
+```bash
+main.w
+wing.toml
+package.json
+```
+
+If we look at the `main.w` file, we'll see:
+
+```jsx
+bring cloud;
+
+let api = new cloud.Api();
+api.get("/", inflight () => {
+  return {
+    status: 200,
+    body: "hello, from within the VPC!"
+  };
+});
+```
+
+You'll notice, there's nothing in this Wing code that implies that the API needs to be inside the VPC. This is because in Wing, this type of configuration happens at that *platform level* and not at the *application level*.
+
+Your platform configuration happens inside `wing.toml`:
+
+```toml
+[tf-aws]
+vpc = "new"
+vpc_apigateway = true
+vpc_lambda = true
+```
+
+This `wing.toml` file sets configuration options for the `tf-aws` platform. The quickstart configures the platform to create a new VPC for this app and put Amazon API Gateways and AWS Lambda functions inside that VPC.
+
+> Built-in Wing platforms such as tf-aws support certain common configuration options (such as private APIs). If you need additional customization, you can always create your own custom platforms and have complete control over how your app is deployed to the cloud.
+>
+
+Before we deploy our app to AWS, let's first check it out in the Wing Simulator:
+
+```bash
+wing it
+```
+
+> If this is the first time you are running the Wing Simulator on your machine, you'll need to sign up with your GitHub credentials so that you can later on setup your app for Wing Previews and use cloud-based development tools like webhook forwarding and app sharing.
+>
+
+Once the Wing Simulator is running, you'll be able to see your API endpoint, invoke it and see the response.
+
+Next, let's deploy this to the cloud:
+
+```
+wing compile -t tf-aws
+terraform -chdir=./target/main.tfaws init
+terraform -chdir=./target/main.tfaws apply
+```
+
+> Make sure to have [Terraform](https://developer.hashicorp.com/terraform/install) installed. The `terraform init` step is only required for the initial deployment.
+
+This command will compile and deploy your Wing application to the AWS account configured in your environment. You'll notice that it will create a new VPC for you with all the desired setup.
+
+But what if I wanted to deploy my app into an existing VPC? It's very common for a VPC to be shared across multiple applications. This can be done by editing your `wing.toml` file like this:
+
+```toml
+[tf-aws]
+vpc = "existing"
+vpc_id = "vpc-288494x"
+vpc_private_subnets = [ "subnet-a112", "subnet-a2x1" ]
```

Comment:
Technically right now the platform only supports 1 public and 1 private subnet. Ill open an issue for supporting multiple but lests change this to 
```suggestion
private_subnet_id =  "subnet-a112"
public_subnet_id = "subnet-a2x1"
```